django/db/models/deletion.py

```diff
@@ -136,8 +136,7 @@ def can_fast_delete(self, objs, from_field=None):
         else:
             return False
         if (signals.pre_delete.has_listeners(model) or
-                signals.post_delete.has_listeners(model) or
-                signals.m2m_changed.has_listeners(model)):
+                signals.post_delete.has_listeners(model)):
```

Comment:
Regarding this change, I checked out 1cd6e04cd4f768bcd4385b75de433d497d938f82 and I couldn't come up with a valid reason for `signals.m2m_changed` to be considered because even at that point it was only fired from `clear()` and `remove()`. I initially suspected it might have initially be required because of changes that landed in late 1.4 or early 1.5 but I couldn't figure out which. At this point I'm pretty confident this was simply thrown in because of the `pre_delete`/`post_delete` nature of the signal.

Reply:
As https://github.com/django/django/pull/6579 shows we don't send `m2m_changed` signals during deletion yet so this is effectively a safe change for now.